engine/src/main/java/org/hibernate/validator/constraintvalidation/HibernateConstraintValidatorInitializationContext.java

```diff
@@ -55,4 +55,14 @@
 	 * @return the tolerance
 	 */
 	Duration getTemporalValidationTolerance();
+
+	/**
+	 * Returns an instance of the specified type or {@code null} if no constraint validator payload of the given type
+	 * has been set.
+	 *
+	 * @param type the type of payload to retrieve
+	 * @return an instance of the specified type or null if no payload of this type has been set
+	 * @since 6.0.8
+	 */
+	<C> C getPayload(Class<C> type);
```

Comment:
I was about to say that it should be named `getConstraintValidatorPayload()` for the sake of consistency with the other new API methods. But actually I'm wondering whether this all should be named "payload" at all. That term already is used in the Bean Validation API (`payload()` on constraints, and also `HibernateConstraintViolation#getDynamicPayload()`.

These existing payloads are exposed on the resulting constraint violation. So my question is should we a) use another name here (if it's about some contextual information meant for the constraint validator information) or b) expose that new payload information on resulting constraint violations as we do for the other payloads (which are scoped to a single constraint and its violations, though).

I feel finding another name might be the best thing to avoid confusion with the existing payload concepts (of course a given validator implementation can still decide to forward this new contextual information as a dynamic constraint payload in emitted violations).

Any thoughts?

Reply:
/cc @gsmet

Reply:
@gunnarmorling well I asked on HipChat and you told me to name it whatever I feel it to be the most appropriate so that's what I did :). Considering we are in the validator context, I think the context is clear enough, that's why I went for the short form. I couldn't think of another payload we would want it here. I don't have a strong opinion though as I hesitated between the consistency and the impression of redundancy. So if you prefer the full name, let's name it that way, I'm +0 on this.

As for the payload term, I think it's the appropriate generic term for this sort of things so I don't think it's an issue to use it there and I don't think it collides with the other payloads as it's a very generic term.

Reply:
> I asked on HipChat and you told me to name it whatever I feel it to be the most appropriate so that's what I did :)

Well, of course I was expecting you to do a wise choice ;)

My concern is that using the same name may lead people to expect that it will automatically be exposed via `getDynamicPayload()`, which won't be the case unless a specific validator implementation decides to do so. The problem is though I can't think of another good name either, apart from yet another "context" flavor.

Reply:
I think payload is a more appropriate term than context.

Let's rename `getPayload()` to `getConstraintValidatorPayload()` and be extra clear in the documentation I still have to write :).

Reply:
One other thing to note is that it's only exposed in the *initialization* context passed to the validator. It's not exposed in the `isValid()` context. That might help with the confusion as only people willing to use the advanced `initialize()` method will notice it.

Reply:
Ok, `getConstraintValidatorPayload()` it is. I see it's all marked as incubating, so we got some liberty to adjust down the road, if needed. Thanks!

Reply:
Alright, @gsmet will you do the renaming when you write the docs?

Reply:
Yes, will do.

Reply:
Alright!